### DISCUSSION TITLE
How to recover my Github account with 2FA

### DISCUSSION THREAD

**Discussion by Pragad** [community/community#23491] [New to GitHub]
<p>Hi, I had enabled 2FA in my Github account. I tried using the Github code from <a href="https://apps.apple.com/us/app/google-authenticator/id388497605" rel="noopener nofollow ugc">Google Authenticator</a> which I had earlier set up. But that doesn’t work.</p>
<p>I have also saved my github-recovery-codes.txt file. This is a multi-line file. When I enter the contents from this file into my Github recover textbox, I see a bunch of spaces that get added. My recover code file is of the below format</p>
<pre><code class="lang-auto">12345-54321
abcde-asdfg
...
</code></pre>
<p>The above recovery code gets converted into <code>12345-54321 abcde-asdfg</code>.</p>
<p>Finally, I tried the last resort. But I am unable to select any of the verification methods.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/4/a483cfe525807d2379421d58cfbe2097cf2ee990.png" data-download-href="/uploads/short-url/ntmI63SkvlYTPiqW9uhIXYGgfNm.png?dl=1" title="Screen Shot 2022-05-20 at 3.43.46 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1150221/181077396-90067e43-9b6e-4395-8e78-2f04971f29f4.png" alt="Screen Shot 2022-05-20 at 3.43.46 PM" data-base62-sha1="ntmI63SkvlYTPiqW9uhIXYGgfNm" width="299" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/4/a483cfe525807d2379421d58cfbe2097cf2ee990_2_299x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/4/a483cfe525807d2379421d58cfbe2097cf2ee990_2_448x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/a/4/a483cfe525807d2379421d58cfbe2097cf2ee990_2_598x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/4/a483cfe525807d2379421d58cfbe2097cf2ee990_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screen Shot 2022-05-20 at 3.43.46 PM</span><span class="informations">750×1252 59.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>I would have rather disabled 2FA in my Github account. Do folks have suggestions on how I can recover my account?</p>

**Comment by ettaboyle** [comment, marked as answer]
<p>:wave:t2: Hi <a class="mention" href="/pragad">@Pragad</a> – Welcome to the forums!! Okay – so here is what I can share. We can’t really answer account-specific questions on the public forum and often your best bet will be to contact GitHub support (open a ticket on our <a href="https://support.github.com/contact">Support page</a>)</p>
<p>BUT … there are a couple of publicly available docs on 2FA that might solve your problem:</p>
<ul>
<li><a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/configuring-two-factor-authentication-recovery-methods">Configuring two-factor authentication recovery methods</a></li>
<li><a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/recovering-your-account-if-you-lose-your-2fa-credentials">Recovering your account if you lose your 2FA credentials</a></li>
</ul>
<p>I hope🤞🏻you can find your solution in one of those. If not, again, your best bet is support.</p>
<p>Thanks for reaching out &amp; good luck!! If you happen to get this solved and you want to close the loop on the thread and share any tips or tricks for others who may encounter a similar problem that would be amazing! :sparkles:</p>
